File: qqbot-websocket/src/main/java/me/zhenxin/qqbot/websocket/entity/Ready.java
/*
 * qq-official-bot-sdk(qqbot) - QQ Official Bot SDK For Java
 * Copyright (C) 2021-2022 xiaoye-bot Project Team
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */

package me.zhenxin.qqbot.websocket.entity;

import lombok.Data;
import me.zhenxin.qqbot.entity.User;

import java.util.List;

/**
 * WS链接成功
 *
 * @author 真心
 * @since 2021/12/11 16:03
 */
@Data
public class Ready {
    private Integer version;
    private String sessionId;
    private User user;
    private List<Integer> shard;
}

File: qqbot-websocket/src/main/kotlin/me/zhenxin/qqbot/entity/Ready.kt
/*
 * QQBot SDK - QQ Official Bot SDK
 * Copyright (C) 2023 ZhenXin
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */

package me.zhenxin.qqbot.entity

import com.alibaba.fastjson2.annotation.JSONField

/**
 * 准备就绪
 *
 * @author 真心
 * @since 2023/10/23 10:57
 */
data class Ready(
    /**
     * 版本
     */
    val version: String,
    /**
     * 会话 ID
     */
    @JSONField(name = "session_id")
    val sessionId: String,
    /**
     * 用户
     */
    val user: User,
    /**
     * 分片数据
     */
    val shard: List<Int>,
)
